src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -198,8 +196,10 @@ private async Task UpdateSymbolTreeInfoAsync(Project project, CancellationToken
                     return;
                 }
 
-                if (!RemoteFeatureOptions.ShouldComputeIndex(project.Solution.Workspace))
+                var client = await project.Solution.Workspace.TryGetRemoteHostClientAsync(cancellationToken).ConfigureAwait(false);
+                if (client != null)
```

Comment:
this feels like a very weird way to do things.  Can we just have a bit saying if we're the local-editing-workspace, vs the global-searching-workspace?

Reply:
I thought I am following the pattern we already have. all other ones basically doing the same thing. if there is no remote host, it does work inproc, if there is, it call the remote to do the work.

here point is using remote host existance to figure out where work should be done.